block/retriever.go

```diff
@@ -62,7 +63,14 @@ func (m *Manager) syncUntilTarget(ctx context.Context, syncTarget uint64) error
 		if err != nil {
 			return err
 		}
+
 	}
+	// check for cached blocks
+	err := m.attemptApplyCachedBlocks(ctx)
```

Comment:
This shouldn't be here. it should be in `applyBlockCallback` function. Not sure when it was removed from there.
 
The purpose of this is the following:
1. full node is at height x
2. full node got gossiped with blocks at height x+2, x+3,.. , x+n 
3. full node can't apply it until it has x+1 recieved, so it keeps those blocks in the cache
4. once full node gets x+1, than it applies all the rest from the cache.

Not sure why was this moved here and what's the purpose of it here.

Reply:
I think it's importnat for it to be here
1. full node missed x
2. full node got x+1 till x+3000
3. got sync target of assuming 2000 from SL

it make sense to go over the cache and apply what possible.
otherwise u'll wait for next gossiped block